### PR TITLE
Update regenie recipe

### PR DESCRIPTION
**Pull request by joellembatchou**
Describe your pull request here

Update to make regenie recipe for latest v3.0.1
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by joellembatchou** [comment]
@BiocondaBot please add label

**Comment by matuskosut** [comment]
@BiocondaBot please fetch artifacts

**Comment by matuskosut** [comment]
It seems that this file is missing:
```
patches/0005-use-conda-cmakelist-file.patch
```

**Review by dpryan79 — DISMISSED** [review]

**Comment by joellembatchou** [comment]
I checked the test_linux log and it says "Ignoring 1 blacklisted recipes (regenie)". I added the patch file though

**Comment by dpryan79** [comment]
I've removed this from the blacklist. I think I ran into issues during the last migration with some dependencies conflicting.

**Comment by joellembatchou** [comment]
@BiocondaBot please add label

File: recipes/regenie/conda_build_config.yaml [hunk 1]
```diff
@@ -1,6 +1,2 @@
 boost_cpp:
-  - 1.63
-  - 1.65
-  - 1.70
-  - 1.72
   - 1.74
```
Comment (dpryan79) [inline recipes/regenie/conda_build_config.yaml:2]: Can you remove this file? boost_cpp is pinned to 1.74 project-wide already.

**Comment by joellembatchou** [comment]
@matuskosut Finished passing all the edits and it looks like it passed all the checks. Let me know if I need to do anything further.

File: recipes/regenie/meta.yaml [hunk 2]
```diff
@@ -1,45 +1,49 @@
...
+    - binutils >=2.33.1  # [linux]
+    - cmake >=3.13
     - make
     - llvm-openmp  # [osx]
     - libgomp  # [linux]
   host:
     - bgenix >=1.1.7
-    - boost-cpp
-    - libblas
+    - boost-cpp {{ boost_cpp }}
```
Comment (dpryan79) [inline recipes/regenie/meta.yaml:38]: You can remove "{{ boost_cpp}}" both here and below. boost-cpp is always pinned to the exact patch version a package was built with.

**Review by dpryan79 — CHANGES_REQUESTED** [review]
A couple small changes and this should be good to go

**Comment by matuskosut** [comment]
@BiocondaBot please fetch artifacts

**Review by matuskosut — APPROVED** [review]
LGTM

**Comment by matuskosut** [comment]
@BiocondaBot please merge

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@BiocondaBot please merge

**Pull request merged**